examples/tls/tls_client.c

```diff
@@ -329,22 +328,30 @@ int TPM2_TLS_ClientArgs(void* userCtx, int argc, char *argv[])
 #endif /* !NO_FILESYSTEM */
 
     /* Client Key (Mutual Authentication) */
-    /* Note: Client will not send a client certificate unless a private key is
-     *   set, so we use a fake "DUMMY" key tell wolfSSL to send certificate.
-     *   The crypto callback will detect use of the dummy key using myTpmCheckKey
+    /* Note: Client will not send a client certificate unless a key is
+     *   set. Since we do not have the private key wolfSSL allows setting a
+     *   public key instead (if crypto callbacks are enabled).
      */
 #ifndef NO_TLS_MUTUAL_AUTH
     if (!useECC) {
     #ifndef NO_RSA
-        printf("Loading RSA dummy key\n");
+        byte der[1024];
+        word32 derSz = sizeof(der);
+        rc = wc_RsaKeyToPublicDer_ex(&wolfRsaKey, der, derSz, 1);
+        if (rc < 0) {
+            printf("Failed to export RSA public key!\n");
+            goto exit;
+        }
+        derSz = rc;
+        rc = 0;
 
         /* Private key is on TPM and crypto dev callbacks are used */
-        /* TLS client (mutual auth) requires a dummy key loaded (workaround) */
-        if (wolfSSL_CTX_use_PrivateKey_buffer(ctx, DUMMY_RSA_KEY,
-            sizeof(DUMMY_RSA_KEY), WOLFSSL_FILETYPE_ASN1) != WOLFSSL_SUCCESS) {
-            printf("Failed to set key!\r\n");
+        /* TLS client (mutual auth) requires a public key loaded */
```

Comment:
The API is for using a private key, so this comment "LOOKS" like a mistake.  Can we add something to say that we're doing this on purpose? Something like "Even though this is API is for a private key, ....."